sign/sign.go

```diff
@@ -168,3 +162,17 @@ func (s *Signer) VerifyFile(path string) (*SignedObject, error) {
 
 	return &SignedObject{}, nil
 }
+
+// enableExperimental sets the cosign experimental mode to true. It also
+// returns a resetFn to recover the original state within the environment.
+func (s *Signer) enableExperimental() (resetFn func(), err error) {
+	const key = "COSIGN_EXPERIMENTAL"
+	if err := s.impl.Setenv(key, "true"); err != nil {
+		return nil, errors.Wrap(err, "enable cosign experimental mode")
+	}
+	return func() {
+		if err := s.impl.Setenv(key, ""); err != nil {
```

Comment:
Does it make sense that we first check what is the original value and reset it to that value? For example, let's say someone uses this library on their local system (for any reason). They might have `COSIGN_EXPERIMENTAL` enabled in their env, and in such a case, it probably doesn't make sense to disable it.

Reply:
I think that is valid in the context of the process, it can get existing env vars and add new ones, but if you create new ones that are valid in the process, outside you don't see that.

I did a small test here and did not see that in the session, it looks like was available only for the process

Reply:
Good point! I changed the implementation to always restore the previous value.